src/command/dev/next/router/config/mod.rs

```diff
@@ -130,44 +131,47 @@ impl RunRouterConfig<RunRouterConfigReadConfig> {
-                        let address = address
-                            .map(RouterAddress::from)
-                            .unwrap_or(self.state.router_address);
-                        let health_check_enabled = router_config.health_check_enabled();
-                        let health_check_endpoint = router_config.health_check_endpoint()?;
-                        let health_check_path = router_config.health_check_path();
-                        let listen_path = router_config.listen_path();
-
-                        Ok(RunRouterConfigFinal {
-                            listen_path,
-                            address,
-                            health_check_enabled,
-                            health_check_endpoint,
-                            health_check_path,
-                            raw_config: contents.to_string(),
-                        })
-                    }
-                    Err(RoverStdError::EmptyFile { .. }) => Ok(RunRouterConfigFinal::default()),
-                    Err(err) => Err(ReadRouterConfigError::ReadFile {
-                        path: path.clone(),
-                        source: Box::new(err),
-                    }),
+            Some(path) => match read_file_impl.read_file(&path).await {
+                Ok(contents) => {
+                    let yaml = serde_yaml::from_str(&contents).map_err(|err| {
+                        ReadRouterConfigError::Deserialization {
+                            path: path.clone(),
+                            source: err,
+                        }
+                    })?;
+
+                    let router_config = RouterConfigParser::new(&yaml);
+                    let address = router_config.address()?;
+                    let address = address
+                        .map(RouterAddress::from)
+                        .unwrap_or(self.state.router_address);
+                    let health_check_enabled = router_config.health_check_enabled();
+                    let health_check_endpoint = router_config.health_check_endpoint()?;
+                    let health_check_path = router_config.health_check_path();
+                    let listen_path = router_config.listen_path();
+
+                    Ok(RunRouterConfigFinal {
+                        listen_path,
+                        address,
+                        health_check_enabled,
+                        health_check_endpoint,
+                        health_check_path,
+                        raw_config: contents.to_string(),
+                    })
                 }
-            }
+                Err(RoverStdError::EmptyFile { .. }) => Ok(RunRouterConfigFinal::default()),
+                Err(RoverStdError::AdhocError { .. }) => {
+                    errln!(
+                        "{} does not exist, creating a router config from CLI options.",
+                        &path
+                    );
+                    Ok(RunRouterConfigFinal::default())
```

Comment:
Here is the new code; rather than assert at the top ```  Fs::assert_path_exists(&path).map_err(ReadRouterConfigError::Fs)?;```, here we'll handle this type of error. This allows us to identify the error, but respond gracefully with the default configuration.